### PR TITLE
Bug/165 - Fix completing timer when paused

### LINKED ISSUE #165: Completed timer not syncing between clients
### Description ###
When the timer is paused and someone completes the timer (Presses the red X), it does not sync between everyone and only resets to 00:00/stops on whoever's timer that clicked the X. Everyone else still sees the timer paused at whatever seconds it was previously at.

### Steps to Reproduce: ###
1. Have at least 2 instances of the timer open.
2. Start timer.
3. Pause timer.
4. Click the red X button.

The timer should now be stopped on the one instance and still paused on the other instance(s).

### PR DESCRIPTION
**Pull request by DigitalEpidemic**
Resolves #165 

In the test `can skip complete timer from websocket message if timer already over`, it was passing in `timerStartedAt: null` and `timerDuration: 0`, but `actions.js` checked if the `timerStartedAt` was `null`. Which happens to also be null when the timer is paused.

This caused pausing the timer then completing it to not sync across clients. Changing it to `timerDuration` seemed to fix the problem for me and not break any existing tests.

### CODE REVIEW

**Comment by mrozbarry** [comment]
Can you reproduce the original issue in a test? This fix looks right, and I want to ensure mobtime has no regressions in the future.

**Comment by DigitalEpidemic** [comment]
I managed to get a test that I think seems to properly test that issue. The only problem is that rather than destructuring `state` and `effect` from `actions.UpdateByWebsocketData()`. I had to just assign it to a single variable called `state` then assert against `state[0]` and `state[1]` for the test runner to give useful differences. If I kept the destructuring, it still passed and failed at the right spots, but only spits out this at any fail:
```
TypeError {
  message: 'undefined is not a function',
}
```
which I don't find useful at all. At least the way I have it gives actual differences. Take a look and let me know what you think!

**Comment by mrozbarry** [comment]
@DigitalEpidemic okay, it looks like I was initially trying to do two things with the code, and both might be reasonably valid.

The first condition to skip completing the timer is `state.timerDuration === 0` like you fixed. There is also `state.timerStartedAt === null`, which is what I think I meant to add before.

The reason both are probably worth checking (using an or) is because `actions.EndTurn` resets both of these values - so if they are both reset, it means the turn has already ended.

In reality, checking either of these conditions should be fine, but checking both may cover an edge case where the duration has reached `0` but hasn't ran the `EndTurn` action yet.

Want to add a second test covering the `state = { timerStartedAt: null }` case? It's okay if you don't, I can add it after a merge. Your call.

File: public/actions.websocket-messages.test.js [hunk 1]
```diff
@@ -123,6 +123,33 @@ test('can complete timer from websocket message', t => {
   );
 });
 
+test('can complete a paused timer from websocket message', t => {
+  const initialState = {
+    timerStartedAt: null,
+    timerDuration: 5000,
+  };
+
+  const state = actions.UpdateByWebsocketData(initialState, {
```
Comment (mrozbarry) [inline public/actions.websocket-messages.test.js:132]: Let's make that `const [state, fx]`

Comment (mrozbarry) [reply public/actions.websocket-messages.test.js:132]: @DigitalEpidemic not sure why you couldn't use `const [state, effect]` here, but I didn't have any issue on my end.

Comment (DigitalEpidemic) [reply public/actions.websocket-messages.test.js:132]: @mrozbarry it lets me use it, but when you actually get the test to fail (like by switching the `state.timerDuration === 0` back to `state.timerStartedAt === null` in the `actions.js` file. The failing error will always be:
```
TypeError {
  message: 'undefined is not a function',
}
```
Rather than comparing the actual output vs the expected output when `state.timerStartedAt === null` is used instead of `state.timerDuration === 0`.

Here is the output of a failing test when using `state[0]` and `state[1]`:
Screen Shot 2020-11-25 at 10 38 09 PM

Here is when `[state, effect]` is used:
Screen Shot 2020-11-25 at 10 38 56 PM

If that doesn't matter, then I'm completely okay with switching back over to the `[state, effect]` way! I was just doing this was cause it felt more useful when needing to debug the failing test.

**Comment by DigitalEpidemic** [comment]
@mrozbarry I see what you mean for checking both conditions, but wouldn't checking with an or `state.timerStartedAt === null || state.timerDuration === 0` go back to not being able to complete the timer when it's paused? Since the paused state seems to set `state.timerStartedAt = null`? Or am I missing something? In my head, I'm thinking there would be another condition in there as well that makes sure that it's skipping the edge case rather than the paused state.

**Comment by mrozbarry** [comment]
@DigitalEpidemic is this bug still present in the current timer?

**Comment by DigitalEpidemic** [comment]
@mrozbarry yes, sir 👀

**Comment by mrozbarry** [comment]
@DigitalEpidemic can you update your fork, and rebase your changes onto master, then I can review. I can do it on my end, but I'd rather you get the credit for the fix if possible.

**Comment by DigitalEpidemic** [comment]
@mrozbarry with the changes made to `websocket.js` ~5 days ago, my fix is no longer valid. Ngl I have no clue how to update the fix to accommodate for those changes.
